build_tools/buildkite/scripts/update_pipeline_configuration.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Updates the YAML steps of a Buildkite pipeline."""
```

Comment:
Overwriting the previously uploaded steps, including any manual edits?

We could wordsmith on the description a bit to make it clear that changes are _not_ made through the UI

![image](https://user-images.githubusercontent.com/4010439/159529420-896ba28e-b7f6-4d98-a98e-a4ffd1c78cb7.png)

(especially since that URL isn't even clickable)

BTW when is that pipeline going to be made public? Also any more thoughts on naming schemes for pipelines? The existing pipelines are `iree-*`, so would we want `iree-presubmit`?

Reply:
There's no reason for it to not be public really. I think the `iree-` prefix is redundant and we shouldn't have it on our new pipelines

Reply:
Made the pipelines public. Which URL are you talking about that isn't clickable?

Reply:
The `https://github.com/google/iree/blob/main/build_tools/buildkit...` URL in the "short pipeline description"